src/gas/fetchFeeHistory.ts

```diff
@@ -0,0 +1,146 @@
+import { BN } from 'ethereumjs-util';
+import { stripHexPrefix } from 'ethjs-util';
+import { query, zipEqualSizedTuples } from '../util';
+
+type HexString = `0x${string}`;
+
+/**
+ * @type FeeHistory
+ *
+ * Historical data for gas and priority fees for a range of blocks within a
+ * particular network. This can be used to gauge how congested the network is.
+ * @property {string} startBlockId - The id of the oldest block in the block
+ * range requested (as hex).
+ * @property {Array<BlockFeeHistory>} blocks - Data for the blocks in the block
+ * range requested.
+ */
+
+type FeeHistory = {
+  startBlockId: HexString;
+  blocks: BlockFeeHistory[];
+};
+
+/**
+ * @type BlockFeeHistory
+ *
+ * Historical data for gas and priority fees for a block.
+ * @property {Array<number>} baseFeePerGas - The base fee per gas for the block.
+ * @property {Array<number>} gasUsedRatio - A number between 0 and 1 that
+ * represents the ratio between the effective gas used and the gas limit for
+ * the block.
+ * @property {Map<number, number>} priorityFeesByPercentile - The priority
+ * fees paid for the transactions in the block that occurred at particular
+ * percentiles (using the gas used vs. gas limit ratio to determine those
+ * percentiles), then indexed by those percentiles.
+ */
+
+type BlockFeeHistory = {
+  baseFeePerGas: number;
+  gasUsedRatio: number;
+  priorityFeesByPercentile: Map<number, number>;
+};
+
+type EthFeeHistoryResponse = {
+  jsonrpc: string;
+  id: number;
+  result: {
+    oldestBlock: HexString;
+    baseFeePerGas: HexString[];
+    gasUsedRatio: number[];
+    reward: HexString[][];
+  };
+};
+
+/**
+ * Converts a hexadecimal string to a decimal number.
+ *
+ * @param hex - A string encoding a hexadecimal number (with a "0x") prefix.
+ * @returns The number in decimal.
+ */
+function hexToDec(hex: string): number {
```

Comment:
I know that we already have some convenience functions around working with hex values in a couple of different places, but it didn't seem like they did exactly what I wanted. For instance, `weiHexToGweiDec` in `util.ts` converts WEI to GWEI, and I don't think we want that in this case. `normalizeGWEIDecimalNumbers` in `gas-util.ts` is better but it ends up converting the final result to a string, and seems like that is unnecessary (if we really wanted a big number, wouldn't we use BN?).

we convert a hex string, we still end up with a string. I really wonder why we are doing this. It seems that if there is a reason for doing so, say, to prevent floating point issues, then BN or BigNumber would be useful in representing those numbers. Otherwise, it does not